src/renderer/sw_engine/tvgSwRenderer.cpp

```diff
@@ -409,6 +409,7 @@ RenderRegion SwRenderer::viewport()
 bool SwRenderer::viewport(const RenderRegion& vp)
 {
     vport = vp;
+    if (surface) vport.intersect({0, 0, (int32_t)surface->w, (int32_t)surface->h});
```

Comment:
If it is the same as before, can we skip it? Called every time the canvas is updated or drawn.
Of course, if the comparison check is an unnecessary operation rather than performing intersect() every time, you can ignore this comment.